rskj-core/src/test/java/co/rsk/peg/RegisterBtcTransactionIT.java

```diff
@@ -0,0 +1,217 @@
+
+        PartialMerkleTree pmtWithTransactions = createValidPmtForTransactions(Collections.singletonList(bitcoinTransaction.getHash()), bridgeConstants.getBtcParams());
+        int btcBlockWithPmtHeight = bridgeConstants.getBtcHeightWhenPegoutTxIndexActivates() + bridgeConstants.getPegoutTxIndexGracePeriodInBtcBlocks();
+        int chainHeight = btcBlockWithPmtHeight + bridgeConstants.getBtc2RskMinimumAcceptableConfirmations();
+        recreateChainFromPmt(btcBlockStoreWithCache, chainHeight, pmtWithTransactions, btcBlockWithPmtHeight, bridgeConstants.getBtcParams());
+
+        bridgeStorageProvider.save();
+
+        BridgeSupport bridgeSupport = getBridgeSupport(bridgeStorageProvider, activationConfig, federationSupport, feePerKbSupport, rskExecutionBlock, btcBlockStoreFactory, track, btcLockSenderProvider);
+
+        Transaction rskTx = getRskTransaction();
+        int activeFederationUtxosSizeBeforeRegisteringTx = federationSupport.getActiveFederationBtcUTXOs().size();
+        org.ethereum.crypto.ECKey key = org.ethereum.crypto.ECKey.fromPublicOnly(btcPublicKey.getPubKey());
+
+        RskAddress receiver = new RskAddress(key.getAddress());
+        co.rsk.core.Coin receiverBalance = track.getBalance(receiver);
+
+        bridgeSupport.registerBtcTransaction(rskTx, bitcoinTransaction.bitcoinSerialize(), btcBlockWithPmtHeight, pmtWithTransactions.bitcoinSerialize());
+
+        bridgeSupport.save();
+        track.commit();
+
+        TransactionOutput output = bitcoinTransaction.getOutput(0);
+        UTXO utxo = getUtxo(bitcoinTransaction, output);
+        List<UTXO> activeFederationUtxosSizeAfterRegisteringTx = federationSupport.getActiveFederationBtcUTXOs();
+
+        assertEquals(activeFederationUtxosSizeBeforeRegisteringTx + 1, activeFederationUtxosSizeAfterRegisteringTx.size());
+        assertEquals(Collections.singletonList(utxo), activeFederationUtxosSizeAfterRegisteringTx);
+        assertEquals(receiverBalance.add(co.rsk.core.Coin.fromBitcoin(btcTransferred)), repository.getBalance(receiver));
+        assertTrue(bridgeStorageProvider.getHeightIfBtcTxhashIsAlreadyProcessed(bitcoinTransaction.getHash()).isPresent());
+        assertEquals(rskExecutionBlock.getNumber(), bridgeStorageProvider.getHeightIfBtcTxhashIsAlreadyProcessed(bitcoinTransaction.getHash()).get());
+    }
+
+    private static UTXO getUtxo(BtcTransaction bitcoinTransaction, TransactionOutput output) {
+        return new UTXO(
+                bitcoinTransaction.getHash(),
+                output.getIndex(),
+                output.getValue(),
+                0,
+                bitcoinTransaction.isCoinBase(),
+                output.getScriptPubKey()
+        );
+    }
+
+    private static Transaction getRskTransaction() {
+        Keccak256 rskTxHash = PegTestUtils.createHash3(1);
+        Transaction rskTx = mock(Transaction.class);
+        when(rskTx.getHash()).thenReturn(rskTxHash);
+        return rskTx;
+    }
```

Comment:
nit: Methods like this can be direcly put in a setUp method under a @BeforeEach annotation.

Reply:
Same as before. I left everything there since there's only one test. For future tests I will re-use the variables so I will extract them to a setUp method